### PR TITLE
2991-use-fs-from-device

### PR DESCRIPTION
**Pull request by joshimoo**
Added the ability to use an existing FS even if it's incompatible with the users choosen FS type. Since we previously did not support user configured FS we need to fallback to the existing FS on the device.

longhorn/longhorn#2991

### CODE REVIEW

**Comment by joshimoo** [comment]
/hold merge 
Please review and let me merge this after my final test run.

File: pkg/volume/volume.go
Comment (shuo-wu) [inline pkg/volume/volume.go:58]: Should we error out if there are partitions detected in the Longhorn volume?

File: pkg/volume/volume.go
Comment (shuo-wu) [inline pkg/volume/volume.go:59]: NIT: Add a warning log here

File: pkg/volume/volume.go
Comment (shuo-wu) [inline pkg/volume/volume.go:58]: Can this input be empty now? If YES, we may need to retain this fallback/default value.

Comment (joshimoo) [reply pkg/volume/volume.go:58]: Already handled as part of the kubernetes mounter.

File: pkg/volume/volume.go
Comment (joshimoo) [reply pkg/volume/volume.go:58]: The mount will either succeed or fail as part of the kubernetes mount call, no need to error here. Since this is just to determine if we need the backwards compatibility.

**Review by innobead — APPROVED** [review]
LGTM

**Comment by joshimoo** [comment]
Tested the new changes, updated image.
joshimoo/longhorn-share-manager:v_20210914

**Comment by innobead** [comment]
@joshimoo please remember the image dependency update in the below repos. 

- longhorn/longhorn chart
- longhorn/longhorn-manager deploy & longhorn-images.txt

**Comment by joshimoo** [comment]
Yep was just waiting for the build to finish, then the release build to finish. Now I can update the dependency commits.

**Pull request merged**
